faunadb/query.py

```diff
@@ -40,6 +43,45 @@ def quote(expr):
   """See the `docs <https://faunadb.com/documentation/queries#basic_forms>`__."""
   return {"quote": expr}
 
+thread_local = local()
```

Comment:
I don't think we can move this inside the function. We need it to persist across multiple function calls, providing a different value to fauna_lambda_var_number for different calls.